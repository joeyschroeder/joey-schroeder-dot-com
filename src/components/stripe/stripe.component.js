import React from 'react';
import styles from './stripe.style.scss';

export const Stripe = () => <div className={styles.root} />;
