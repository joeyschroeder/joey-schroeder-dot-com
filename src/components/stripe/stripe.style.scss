@import "../../styles/config/variables";

@keyframes stripeMove {
  0% {
    transform: translateX(-100%);
  }

  100% {
    transform: translateX(0%);
  }
}

.root {
  animation-duration: $duration-base * 7;
  animation-fill-mode: both;
  animation-iteration-count: 1;
  animation-name: stripeMove;
  animation-timing-function: ease;
  background: $color-stripe-5;
  height: $spacer * 2;

  @for $i from 1 through 5 {
    &:nth-child(#{$i}) {
      animation-delay: $i * $duration-base / 3;
    }
  }

  &:nth-child(1) {
    background: $color-stripe-1;
  }

  &:nth-child(2) {
    background: $color-stripe-2;
  }

  &:nth-child(3) {
    background: $color-stripe-3;
  }

  &:nth-child(4) {
    background: $color-stripe-4;
  }
}
