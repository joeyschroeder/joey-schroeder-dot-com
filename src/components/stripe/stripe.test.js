import React from 'react';
import { create } from 'react-test-renderer';
import { Stripe } from './stripe.component';

describe('<Stripe />', () => {
  it('should render correctly', () => {
    const component = create(<Stripe />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
