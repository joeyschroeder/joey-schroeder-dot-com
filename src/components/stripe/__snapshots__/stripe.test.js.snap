// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stripe /> should render correctly 1`] = `
<div
  className="root"
/>
`;
